lib/arel/visitors/oracle12.rb

```diff
@@ -0,0 +1,53 @@
+module Arel
+  module Visitors
+    class Oracle12 < Arel::Visitors::ToSql
+      private
+
+      def visit_Arel_Nodes_SelectStatement o, collector
+        # Oracle does not allow LIMIT clause with select for update
+        if o.limit && o.lock
+          o = o.dup
+          o.limit = []
```

Comment:
Thanks for the comments.  Considered again this behavior and agreed.  Opened https://github.com/rails/arel/pull/430 to change this behavior.